sentinelsat/sentinel.py

```diff
@@ -192,7 +192,8 @@ def load_query(self, query, start_row=0):
 
         # repeat query until all results have been loaded
         if total_results > self.page_size + start_row - 1:
-            self.load_query(query, start_row=(start_row + self.page_size))
+            self.load_query(query, start_row=(start_row + self.page_size), products=output)
```

Comment:
Do `entries += self.load_query(...)` here and `return entries` on the next line.
This line is also the cause of the `test_large_query` failing, since the recursive call does not modify the `output` variable anymore.